app/gui/view/component-browser/breadcrumbs/src/entry.rs

```diff
@@ -0,0 +1,343 @@
+        let frp = EntryFrp::<Self>::new();
+        let input = &frp.private().input;
+        let out = &frp.private().output;
+        let network = frp.network();
+        let color_anim = Animation::new(network);
+        let appear_anim = Animation::new(network);
+        fn mix(c1: &color::Rgba, c2: &color::Rgba, coefficient: &f32) -> color::Rgba {
+            color::mix(*c1, *c2, *coefficient)
+        }
+
+        enso_frp::extend! { network
+            init <- source_();
+            size <- input.set_size.on_change();
+            margin <- input.set_params.map(|p| p.margin).on_change();
+            hover_color <- input.set_params.map(|p| p.hover_color).on_change();
+            font <- input.set_params.map(|p| p.font.clone_ref()).on_change();
+            text_offset <- input.set_params.map(|p| p.text_offset).on_change();
+            text_color <- input.set_params.map(|p| p.selected_color).on_change();
+            text_size <- input.set_params.map(|p| p.text_size).on_change();
+            greyed_out_color <- input.set_params.map(|p| p.greyed_out_color).on_change();
+            greyed_out_from <- input.set_params.map(|p| p.greyed_out_start).on_change();
+            highlight_corners_radius <- input.set_params.map(|p| p.highlight_corners_radius).on_change();
+            transparent_color <- init.constant(color::Rgba::transparent());
+
+            col <- input.set_location._1();
+            should_grey_out <- all_with(&col, &greyed_out_from,
+                |col, from| from.map_or(false, |from| *col >= from)
+            );
+            color_anim.target <+ should_grey_out.map(|should| if *should { 1.0 } else { 0.0 });
+            target_color <- all_with3(&text_color, &greyed_out_color, &color_anim.value, mix);
+            appear_anim.target <+ init.constant(1.0);
+            model_was_set <- input.set_model.map(f!((model) data.is_state_switch(model))).on_true();
+            should_appear <- any(&init, &model_was_set);
+            eval_ should_appear({
+                appear_anim.target.emit(0.0);
+                appear_anim.skip.emit(());
+                appear_anim.target.emit(1.0);
+            });
+            color <- all_with3(&transparent_color, &target_color, &appear_anim.value, mix);
+
+            contour <- all_with(&size, &margin, |size, margin| Contour {
+                size: *size - Vector2(*margin, *margin) * 2.0,
+                corners_radius: 0.0,
+            });
+            layout <- all(contour, text_size, text_offset);
+            eval layout ((&(c, ts, to)) data.update_layout(c, ts, to));
+            eval color((c) data.set_default_color(*c));
+            eval font((f) data.set_font(f.to_string()));
+            eval text_size((s) data.set_default_text_size(*s));
+            is_disabled <- input.set_model.map(|m| matches!(m, Model::Separator | Model::Ellipsis));
+            width <- map2(&input.set_model, &text_offset,
+                f!([data](model, text_offset) {
```

Comment:
you dont need `[data]` here. If the code starts with `something.` then `something` is automatically put there in `f!` macro.